tools/roslaunch/src/roslaunch/loader.py

```diff
@@ -479,7 +479,7 @@ def param_value(self, verbose, name, ptype, value, textfile, binfile, command):
                 from xmlrpc.client import Binary
             except ImportError:
                 from xmlrpclib import Binary
-            with open(binfile, 'rb') as f:
+            with open(binfile, 'r') as f:
```

Comment:
The change looks wrong to me. The `Binary` constructor expects `bytes` and therefore the file is opened with `rb`. When opening the file with only `r` the read data will be of type `str` and not `bytes`.

Reply:
thanks for sharing the expertise! did not know that there is the `bytes` expectation.

the motivation of this change is because of one failure in `test_roslaunch` (https://github.com/ros/ros_comm/blob/melodic-devel/test/test_roslaunch/test/params_basic.py).

In this test case, binary data is read from files with format specified as [`test_binary_data = open('example.launch', 'rb').read()`](https://github.com/ros/ros_comm/blob/melodic-devel/test/test_roslaunch/test/params_basic.py#L111), and compared against `get_param("commandoutput")`

however, the latter one (`get_param("commandoutput")`) is interpreted as `Binary(open(binfile, 'rb').read())` (ignoring the change in this pull request), leading to failure during comparison.